src/webots/nodes/WbTransform.cpp

```diff
@@ -406,3 +406,39 @@ QStringList WbTransform::fieldsToSynchronizeWithX3D() const {
          << "rotation";
   return fields;
 }
+
+WbVector3 WbTransform::translationFrom(WbNode *fromNode) const {
+  const WbNode *parentNode = parent();
+  const WbNode *childNode = this;
+  WbVector3 translationResult;
+
+  while (parentNode != fromNode) {
+    WbVector3 translation;
+    if (dynamic_cast<const WbTransform *>(childNode))
+      translation = static_cast<const WbTransform *>(childNode)->translation();
+    translationResult += translation;
+
+    childNode = parentNode;
+    parentNode = parentNode->parent();
```

Comment:
The `WbNodeUtilities::findUpperTransform(const WbNode *node)` function might simplify this function.

Reply:
Nice, it works!